SchedulePlatform.Api/Controllers/AppointmentController.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+using SchedulePlatform.Api.Models;
+using SchedulePlatform.Api.Mappings;
+using SchedulePlatform.Models.Entities;
+using SchedulePlatform.Service.Interfaces;
+
+namespace SchedulePlatform.Api.Controllers
+{
+	[Route("api/[controller]")]
+
+	[ApiController]
+
+	public class AppointmentController : ControllerBase
+	{
+		private readonly IAppointmentService _service;
+
+		public AppointmentController(IAppointmentService service)
+		{
+			_service = service;
+		}
+
+		[HttpGet]
+
+		public Appointment[] GetAppointments()
+		{
+			return _service.GetAll();
+		}
+
+		[HttpPost]
+
+		public Appointment AddAppointment(Appointment appointment)
+		{
+			return _service.Add(appointment);
+		}
+
+		[HttpGet("{id}")]
+
+		public IActionResult GetById(Guid id)
+		{
+			var appointmentResult = _service.GetById(id);
+
+			if (appointmentResult==null)
+			{
+				return NotFound();
+			}
+			return Ok(appointmentResult);
+		}
+
+		[HttpPatch]
+
+		public IActionResult Update(Guid id, AppointmentPatchModel model)
+		{
+			var appointmentSearch = _service.GetById(id);
+
+			if (appointmentSearch==null)
+			{
+				return NotFound();
+			}
+
+			var appointmentUpdated = appointmentSearch.Map(model);
```

Comment:
Same here. Lines 56 to 61 should be in a service. It's not a controller's responsibility to do checks and mappings because they imply logic so they should sit in the service.